### DISCUSSION TITLE
Dynamoose with Annotations!

### DISCUSSION THREAD

**Discussion by changmyeong** [dynamoose/dynamoose#1656] [Announcements]
Hello guys,

We developed dynamoose-decorator to make Dynamoose easy to use in TypeScript.
We’d like to receive feedback!

https://github.com/p1ayground/dynamoose-decorator

<img width="522" alt="스크린샷 2024-01-21 오전 4 34 06" src="https://github.com/dynamoose/dynamoose/assets/21070153/4fba3fa9-35be-415b-8cfd-803b6d1f22b4">

**Comment by fishcharlie** [comment]
@changmyeong Wow! Really impressive. Would you be interested in trying to create a PR and get this into Dynamoose itself? 😃

**Reply by changmyeong** [reply]
Sure! But I think the task will be big.
First of all, it might be a good idea to add the dynamoose-decorator repository to your TypeScript guide to adapt the interface as it gets popular and for extensive testing.

https://dynamoosejs.com/getting_started/TypeScript

**Reply by changmyeong** [reply]
Let's do as we do in sequelize and plan to merge into dynamoose.
https://sequelize.org/docs/v6/other-topics/typescript/

**Reply by fishcharlie** [reply]
@changmyeong Yeah I was just thinking about that. I'm trying to decide if it would be better to create a new page for Dynamoose 3rd party libraries/resources. Or if I should link to it directly from the TypeScript page.

All the markdown files for the website are here if you wanna submit a PR: https://github.com/dynamoose/dynamoose/tree/main/docs/docs_src

**Reply by changmyeong** [reply]
Okay. We'll submit a PR as quickly as possible :)

**Reply by changmyeong** [reply]
@fishcharlie We submited a PR! https://github.com/dynamoose/dynamoose/pull/1657

**Reply by leandroluk** [reply]
I had created a fullish wrapper over dynamoose like TypeORM here: https://www.npmjs.com/package/@leandroluk/dynamoose-typed

**Comment by leandroluk** [comment]
Take a look on https://www.npmjs.com/package/@leandroluk/dynamoose-typed
